File: Assignment_43/Program3.c
/*
Write a recursive program which accept string from user and count number of small characters
Input : HElloWOrlD
Output : 5
*/

#include <stdio.h>

int Small(char *str)
{
    static int iCount = 0;

    if(*str != '\0')
    {
       if((*str >= 'a') && (*str <= 'z'))
        {
            iCount++;
        }
        str++;
        Small(str);
    }
    return iCount;
}

int main()
{
    int iRet = 0;
    char arr[20] = {'\0'};

    printf("Enter string : ");
    scanf("%s",arr);

    iRet = Small(arr);

    printf("%d",iRet);

    return 0;
}

File: Assignment_54/Program3.java
/* Write a java program which accept array of characters from user & count number of capital characters.
Input :   b N j B R b A d G G
Output :  6
*/

import java.util.*;

class MyArray
{
    public int ArrayCapital(char arr[])
    {
        int i = 0,iCount = 0;
        for (i = 0;i < arr.length;i++)
        {
            if (arr[i] >= 'A' && arr[i] <= 'Z')
            {
                iCount++;
            }
        }
        return iCount;
    }
}

class Program3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int i = 0;

        System.out.println("Enter characters in the array :");
        String line = sobj.nextLine();
        String[] tokens = line.trim().split("\\s+");
        int size = tokens.length;
        char arr[] = new char[size];
        for (i = 0;i < size;i++)
        {
            arr[i] = arr[i] = tokens[i].charAt(0);
        }

        MyArray mobj = new MyArray();

        int iCount = mobj.ArrayCapital(arr);

        System.out.println("Number of capital characters are : "+iCount);

        sobj.close();
    }
}